### ISSUE TITLE
--out with sys.doc always falls back to nested

### ISSUE THREAD

**Issue by sambev** [saltstack/salt#34425]
### Description of Issue/Question
When trying to run something like `salt * sys.doc <anything> --out=json` I do not get json. Doesn't seem to be working with other outputters either

### Setup
Have a minion and a master.

### Steps to Reproduce Issue
```
salt minion1 sys.doc test.ping --out=json
"test.ping:"

    Used to make sure the minion is up and responding. Not an ICMP ping.

    Returns ``True``.

    CLI Example:

        salt '*' test.ping
```

`salt-call` on the minion works:
```
salt-call sys.doc test.ping --out=json
{
    "local": {
        "test.ping": "\n    Used to make sure the minion is up and responding. Not an ICMP ping.\n\n    Returns ``True``.\n\n    CLI Example:\n\n        salt '*' test.ping\n    "
    }
}
```

### Versions Report
Master:
```
Salt Version:
           Salt: 2016.3.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.2

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-32-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

Minion:
```
Salt Version:
           Salt: 2016.3.1

Dependency Versions:
           cffi: 1.2.1
       cherrypy: 3.6.0
       dateutil: 2.4.2
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.22
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.10 (default, Oct 23 2015, 19:19:21)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist:
        machine: x86_64
        release: 15.5.0
         system: Darwin
        version: 10.11.5 x86_64
```

**Comment by Ch3LL** [comment]
Also seeing this behavior in 2015.8.10 and 2015.5.10 as well.

**Comment by rallytime** [comment]
@sambev This should be fixed with #35286 and will be available in the 2016.3.3 release.
